plugin-server/src/worker/ingestion/person-state.ts

```diff
@@ -323,15 +323,15 @@ export class PersonState {
             return undefined
         }
         if (isDistinctIdIllegal(mergeIntoDistinctId)) {
-            await captureIngestionWarning(this.db, teamId, 'cannot_merge_with_illegal_distinct_id', {
+            await captureIngestionWarning(this.db.kafkaProducer, teamId, 'cannot_merge_with_illegal_distinct_id', {
                 illegalDistinctId: mergeIntoDistinctId,
                 otherDistinctId: otherPersonDistinctId,
                 eventUuid: this.event.uuid,
             })
```

Comment:
I think it's best not to rate-limit these, as they contain useful info and merges are supposed to be rare.
I support not making the rate-limiting optional to avoid deploying a bomb, but let's add `mergeIntoDistinctId` as the debounce key for these the three warnings on this file

Reply:
pushed a solution to let us override the debounce key construction so that the three uses in the person state are debounced together.... is that what you meant?

(happy to change it if I'm being silly :))

Reply:
Sorry for being unclear: I do not wish to debounce these warnings, and send all of them to CH. They should be extremely infrequent unless there's a subtle instrumentation bug, and in that case it's better to have as much info as possible.

Reply:
so keeping the safe by default... i've made it opt in to always sending rather than implicit on absent debounce key for e.g.

so

1. always debounce on team id and type
2. unless you provide a debounce key in which case it is team id, type, and key
3. unless you say alwaysSend in which case it always sends